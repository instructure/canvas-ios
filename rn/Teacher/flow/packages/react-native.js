//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
