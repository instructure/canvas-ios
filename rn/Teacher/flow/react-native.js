//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
