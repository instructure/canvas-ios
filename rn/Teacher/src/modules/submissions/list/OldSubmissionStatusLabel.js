//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow
import React, { Component } from 'react'
import {
  StyleSheet,
} from 'react-native'
import i18n from 'format-message'
import { Text } from '../../../common/text'
import { colors } from '../../../common/stylesheet'

type OldSubmissionStatusLabelProps = {
  status: SubmissionStatus,
  onlineSubmissionType?: boolean,
  style?: any,
}

export default class OldSubmissionStatusLabel extends Component<OldSubmissionStatusLabelProps, any> {
  render () {
    let color: string = colors.textDark // none
    let title: string = i18n('Not Submitted')

    switch (this.props.status) {
      case 'late':
        color = colors.textWarning
        title = i18n('Late')
        break
      case 'missing':
        color = colors.textDanger
        title = i18n('Missing')
        break
      case 'submitted':
      case 'resubmitted':
        color = colors.textSuccess
        title = i18n('Submitted')
        break
      case 'excused':
        title = i18n('Excused')
        break
    }

    if (this.props.onlineSubmissionType !== undefined && !this.props.onlineSubmissionType) { title = '' }

    return <Text style={[styles.statusText, this.props.style, { color }]}>{title}</Text>
  }
}

const styles = StyleSheet.create({
  statusText: {
    fontSize: 14,
    paddingTop: 2,
  },
})
