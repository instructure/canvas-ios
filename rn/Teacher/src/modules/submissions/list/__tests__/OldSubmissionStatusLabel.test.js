//
// This file is part of Canvas.
// Copyright (C) 2019-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import 'react-native'
import React from 'react'
import OldSubmissionStatusLabel from '../OldSubmissionStatusLabel'
import renderer from 'react-test-renderer'

describe('SubmissionStatus', () => {
  it('status `none` renders properly', () => {
    let tree = renderer.create(
      <OldSubmissionStatusLabel status={'none'} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('status `missing` renders properly', () => {
    let tree = renderer.create(
      <OldSubmissionStatusLabel status={'missing'} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('status `late` renders properly', () => {
    let tree = renderer.create(
      <OldSubmissionStatusLabel status={'late'} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('status `submitted` renders properly', () => {
    let tree = renderer.create(
      <OldSubmissionStatusLabel status={'submitted'} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('status `excused` renders properly', () => {
    let tree = renderer.create(
      <OldSubmissionStatusLabel status='excused' />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
