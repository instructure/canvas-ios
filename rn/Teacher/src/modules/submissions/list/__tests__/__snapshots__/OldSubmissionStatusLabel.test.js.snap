// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionStatus status \`excused\` renders properly 1`] = `
<Text
  accessibilityLabel="Excused,"
  style={
    Array [
      Object {
        "color": "#2D3B45",
        "fontFamily": "Lato-Regular",
        "fontSize": 16,
      },
      Array [
        Object {
          "fontSize": 14,
          "paddingTop": 2,
        },
        undefined,
        Object {
          "color": "#556572",
        },
      ],
    ]
  }
>
  Excused
</Text>
`;

exports[`SubmissionStatus status \`late\` renders properly 1`] = `
<Text
  accessibilityLabel="Late,"
  style={
    Array [
      Object {
        "color": "#2D3B45",
        "fontFamily": "Lato-Regular",
        "fontSize": 16,
      },
      Array [
        Object {
          "fontSize": 14,
          "paddingTop": 2,
        },
        undefined,
        Object {
          "color": "#FC5E13",
        },
      ],
    ]
  }
>
  Late
</Text>
`;

exports[`SubmissionStatus status \`missing\` renders properly 1`] = `
<Text
  accessibilityLabel="Missing,"
  style={
    Array [
      Object {
        "color": "#2D3B45",
        "fontFamily": "Lato-Regular",
        "fontSize": 16,
      },
      Array [
        Object {
          "fontSize": 14,
          "paddingTop": 2,
        },
        undefined,
        Object {
          "color": "#EE0612",
        },
      ],
    ]
  }
>
  Missing
</Text>
`;

exports[`SubmissionStatus status \`none\` renders properly 1`] = `
<Text
  accessibilityLabel="Not Submitted,"
  style={
    Array [
      Object {
        "color": "#2D3B45",
        "fontFamily": "Lato-Regular",
        "fontSize": 16,
      },
      Array [
        Object {
          "fontSize": 14,
          "paddingTop": 2,
        },
        undefined,
        Object {
          "color": "#556572",
        },
      ],
    ]
  }
>
  Not Submitted
</Text>
`;

exports[`SubmissionStatus status \`submitted\` renders properly 1`] = `
<Text
  accessibilityLabel="Submitted,"
  style={
    Array [
      Object {
        "color": "#2D3B45",
        "fontFamily": "Lato-Regular",
        "fontSize": 16,
      },
      Array [
        Object {
          "fontSize": 14,
          "paddingTop": 2,
        },
        undefined,
        Object {
          "color": "#127A1B",
        },
      ],
    ]
  }
>
  Submitted
</Text>
`;
